### ISSUE TITLE
Android back button results in blank screen: can't block with popscope

### ISSUE THREAD

**Issue by BarryCarlyon** [jb3rndt/PersistentBottomNavBarV2#139]
Package Version (from pubspec.yaml): `persistent_bottom_nav_bar: ^5.0.0`

**Bug Description**

If you navigate backwards from the home page of an app using the android back button you get black screen
Press it again goes to another app/home page as expected.

The black screen shouldn't exist

Additionally, `PopScope` would be used to block the back button functioning and it doesn't when you wrap persistennav in a PopScope

**To Reproduce**

1. Launch you app
2. Use the Android back button, either hardward or software versions
3. Be pressnted with a blank page

This can result in an error message/console error messages as it's a totally unexpected state

as per my rubber ducking in https://github.com/jb3rndt/PersistentBottomNavBarV2/issues/137#issuecomment-2032944561

**Expected behavior**

- PopScope to work to prevent back "leaving" the app.
- If you navigate back without popscope it goes where it's supposed to not a blank page

**Code to reproduce**

This will block back working from the home page
```dart
    return ChangeNotifierProvider(
        create: (context) => ApiService(),
        child: PopScope(
            canPop: false,
            child: HomePageBuild()
        )
    );
```

This does not:

`HomePageBuild` in this case is my first tab in buildTabs.

```dart
    return ChangeNotifierProvider(
        create: (context) => ApiService(),
        child: PopScope(
            canPop: false,
            child: PersistentTabView(
          tabs: _buildTabs(),

          navBarOverlap: const NavBarOverlap.none(),
          navBarBuilder: (navBarConfig) => Style3BottomNavBar(
            navBarConfig: navBarConfig,
            navBarDecoration: const NavBarDecoration(
              color: superdarkRed,
            ),
          ),
          backgroundColor: superdarkRed,
        )
      )
    );
```

With or without popscope the blank page occurs

```dart
    return ChangeNotifierProvider(
      create: (context) => ApiService(),
      child: PersistentTabView(
        tabs: _buildTabs(),

        navBarOverlap: const NavBarOverlap.none(),
        navBarBuilder: (navBarConfig) => Style3BottomNavBar(
          navBarConfig: navBarConfig,
          navBarDecoration: const NavBarDecoration(
            color: superdarkRed,
          ),
        ),
        backgroundColor: superdarkRed,
      )
    );
```

**Screenshots**

After hitting the back soft key on the android "remote control" so a physical pixel 6 device here

<img width="476" alt="image" src="https://github.com/jb3rndt/PersistentBottomNavBarV2/assets/20999/aa8fb231-5ab3-4189-af8e-93ab2bc35939">

If I hot reload/hit sync files I can force this error, if I'm on the blank page.

<img width="422" alt="image" src="https://github.com/jb3rndt/PersistentBottomNavBarV2/assets/20999/853bf720-7233-49f3-a5c6-01f6461bc536">

**Comment by BarryCarlyon** [comment]
I added:

```dart
    PersistentTabController _controller = PersistentTabController(initialIndex: 0);
    _controller.addListener(() {
      debugPrint('we be navin');
    });

snip

      child: PersistentTabView(
        controller: _controller,

```

And I don't get a "we be naving" when we go to the blank screen

**Comment by jb3rndt** [comment]
Hi, thank you for providing all the information. I'm pretty sure I finally found the issue. When migrating the internal `WillPopScope` to `PopScope`, the final navigator pop (which causes the black page) does not minimize the app like it used to when returning true from `WillPopScope`.
I just released version 5.2.0 which should contain the fix. Now, a PopScope with canPop: false that wraps the `PersistentTabView` should also work.
Feel free to check it out, I will keep this open for now in case there are still issues with that.
